proxy/src/ctx/transport.rs

```diff
@@ -56,6 +56,29 @@ impl Server {
 
         Arc::new(s)
     }
+
+    pub fn orig_dst_if_not_local(&self) -> Option<SocketAddr> {
+        match self.orig_dst {
+            None => None,
+            Some(orig_dst) => {
+                // If the original destination is actually the listening socket,
+                // we don't want to create a loop.
+                if same_addr(&orig_dst, &self.local) {
```

Comment:
I'm pretty sure this is right but want to confirm: `local` is the _bound_ local addr and not the "listen addr".  i.e. the listen addr might be 0.0.0.0:1234 but the bound addr is going to be i.e. 127.0.0.1:1234

We'll only know the bound addr once we have a socket on the serverside, whereas the local addr is known when we call _listen_ on the server.

Reply:
yes this is right:
```rust
        let local_addr = connection.local_addr().unwrap_or(self.listen_addr);
```

Reply:
I moved this code from `Outbound`, but will need to re-read to make sure.